service/dap/server.go

```diff
@@ -1042,9 +1198,10 @@ func (s *Server) doStepCommand(command string, threadId int) {
 		stopped.Body.Reason = "error"
 		stopped.Body.Text = err.Error()
 		s.send(stopped)
+		s.asyncCommandDone(asyncSetupDone)
```

Comment:
how about calling this as the very first one after if err != nil so even when s.send panics, s.asyncCommandDone runs and unblocks the handler loop?

Reply:
Oh darn panics. I keep forgetting about them. In that case its safest to just defer. And then we handle the panic from SwitchGoroutine as well. If it's already unblocked, it will do nothing.